### PR TITLE
task/local_db/refactor_processor

### PR DESCRIPTION
**Pull request by teamagam5**
moved logics to be after repositories

### CODE REVIEW

File: domain/src/main/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractor.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    DataSubscriptionRequest dataSubscriptionRequest =
+        factory.create(mVectorLayersRepository.getVectorLayersObservable(),
+            vlObservable -> vlObservable.map(
+                this::cacheLayer) // mapping in order to wait for the caching procedure
+                .doOnNext(this::setVisible)
+                .doOnNext(this::alertIfNeeded)
+                .retryWhen(new RetryWithDelay(Constants.LAYER_CACHING_RETRIES,
+                    Constants.LAYER_CACHING_RETRIES_DELAY_MS))
+                .doOnError(this::logFailure)
+                .onErrorResumeNext(Observable.empty()));
```
Comment (BarAgam) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractor.java]: We need to see that now this line doesn't prevent the processing of future vector layers

**Review by BarAgam — APPROVED** [review]
Bravo

File: domain/src/main/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractor.java (hunk 1 above)
Comment (teamagam5) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractor.java]: If you'd have known how much trouble you caused by this little cute observation... Blimey!

Comment (teamagam5) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractor.java]: PLS re-CR the last commit

File: domain/src/test/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractorCachingTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  public void whenNonCachedVectorLayerEmitted_cacheIt() throws Exception {
+    // Act
+    executeInteractor();
+
+    // Assert
+    Mockito.verify(mCache).cache(mVectorLayer1);
+  }
+
+  @Test
+  public void whenCannotCacheVectorLayer_InteractorIgnoresItAndContinues() throws Exception {
```
Comment (BarAgam) [inline domain/src/test/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractorCachingTest.java:52]: <3

**Review by BarAgam — APPROVED** [review]

**Pull request merged**